TESTS/network/multihoming/main.cpp

```diff
@@ -0,0 +1,169 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#define WIFI 2
+#if !defined(MBED_CONF_TARGET_NETWORK_DEFAULT_INTERFACE_TYPE) || \
+    (MBED_CONF_TARGET_NETWORK_DEFAULT_INTERFACE_TYPE == WIFI && !defined(MBED_CONF_NSAPI_DEFAULT_WIFI_SSID))
+#error [NOT_SUPPORTED] No network configuration found for this target.
+#endif
+#ifndef MBED_CONF_APP_ECHO_SERVER_ADDR
+#error [NOT_SUPPORTED] Requires parameters from mbed_app.json
+#endif
+
+#include "mbed.h"
+#include "greentea-client/test_env.h"
+#include "unity/unity.h"
+#include "utest.h"
+#include "utest/utest_stack_trace.h"
+#include "multihoming_tests.h"
+
+using namespace utest::v1;
+
+namespace {
+NetworkInterface *net;
+}
+
+char interface_name[MBED_CONF_MULTIHOMING_MAX_INTERFACES_NUM][INTERFACE_NAME_LEN];
+int  interface_num = 0;
+
+#if MBED_CONF_NSAPI_SOCKET_STATS_ENABLE
+mbed_stats_socket_t udp_stats[MBED_CONF_NSAPI_SOCKET_STATS_MAX_COUNT];
+#endif
+
+#if defined(MBED_CONF_APP_WIFI_SECURE_SSID) || defined(MBED_CONF_APP_WIFI_UNSECURE_SSID)
+#define SSID_MAX_LEN 32
+#define PWD_MAX_LEN 64
+
+WiFiInterface *wifi;
+#endif
+
+NetworkInterface *get_interface()
+{
+    return net;
+}
+
+static void _ifup()
+{
+
+#if DEVICE_EMAC
+    net = EthInterface::get_default_instance();
+    nsapi_error_t err = net->connect();
+    net->get_interface_name(interface_name[0]);
+    interface_num++;
+    TEST_ASSERT_EQUAL(NSAPI_ERROR_OK, err);
+    printf("MBED: UDPClient IP address is '%s' interface name %s\n", net->get_ip_address(), interface_name[0]);
+#endif
+#if defined(MBED_CONF_APP_WIFI_SECURE_SSID) || defined(MBED_CONF_APP_WIFI_UNSECURE_SSID)
+    wifi = WiFiInterface::get_default_instance();
+
+    if (wifi) {
+#if defined(MBED_CONF_APP_WIFI_SECURE_SSID)
+        char ssid[SSID_MAX_LEN + 1] = MBED_CONF_APP_WIFI_SECURE_SSID;
+        char pwd[PWD_MAX_LEN + 1] = MBED_CONF_APP_WIFI_PASSWORD;
+        nsapi_security_t security = NSAPI_SECURITY_WPA_WPA2;
+
+#elif defined(MBED_CONF_APP_WIFI_UNSECURE_SSID)
+        char ssid[SSID_MAX_LEN + 1] = MBED_CONF_APP_WIFI_UNSECURE_SSID;
+        char pwd[PWD_MAX_LEN + 1] = NULL;
+        nsapi_security_t security = NSAPI_SECURITY_NONE;
+#endif
+
+        printf("\nConnecting to %s...\n", ssid);
+        int ret = wifi->connect(ssid, pwd, security);
+        if (ret != 0) {
+            printf("\nConnection error: %d\n", ret);
+            return;
+        }
+        wifi->get_interface_name(interface_name[1]);
+        interface_num++;
+        printf("MAC: %s\n", wifi->get_mac_address());
+        printf("IP: %s\n", wifi->get_ip_address());
+        printf("Netmask: %s\n", wifi->get_netmask());
+        printf("Gateway: %s\n", wifi->get_gateway());
+        printf("RSSI: %d\n\n", wifi->get_rssi());
+        printf("Wifi interface name: %s\n\n", interface_name[1]);
+
+    } else {
+        printf("ERROR: No WiFiInterface found.\n");
```

Comment:
This must be test error.
You can use `TEST_FAIL_MESSAGE()` macro for that purpose.